### PR TITLE
docs: Rough draft of troubleshooting for apps

### LINKED ISSUE #29989: Include troubleshooting information in agent service Getting Started guides
## Applies To

`/docs/*-access/getting-started/`, except for `/docs/database-access/getting-started/`. This includes the Server Access, Application Access, Kubernetes Access, and Desktop Access guides.

## Details

Add a "Troubleshooting" section to each guide with common errors that users might face even though they have:
- Set up the Prerequisites properly
- Followed each step in the guide

## How will we know this is resolved?

Each guide listed above includes a "Troubleshooting" section.

### PR DESCRIPTION
**Pull request by lsgunn-teleport**
Rough draft based on info provided by @webvictim 

> Application access
> 
> The most common issue I see is CSRF/CORS or incorrect Origin/Host type issues. This comes up most often with Grafana, ArgoCD and many other modern web applications which make heavy use of websockets where cross-site scripting restrictions in browsers prevent traffic from flowing unless explicitly allowed.
> 
> This error usually manifests as lack of functionality and/or displayed errors in the application itself saying that traffic isn't being permitted. Application logs will also often mention CORS or CSRF. 
> 
> The fix is to add forced rewrites for the Origin and Host headers to each Teleport app configuration as per the example below:
> 
```
app_service:
  enabled: true
  apps:
  - name: grafana
    uri: http://localhost:3000/
    public_addr: [grafana.teleport.example.com](http://grafana.teleport.example.com/)
    rewrite:
      headers:
        Origin: "https://grafana.teleport.example.com/" # should be the Teleport application subdomain prepended with "https://"
        Host: "[grafana.teleport.example.com](http://grafana.teleport.example.com/)" # should be the Teleport application subdomain itself
```
In 99% of cases this fixes the CSRF issues and makes the applications work properly.

### CODE REVIEW

File: docs/pages/application-access/troubleshooting-apps.mdx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+traffic isn't being permitted, or application logs that indicate CORS or CSRF errors. 
+
+### Solution
+
+In most cases, you can fix these types of issues by adding explicit `rewrite` settings for the Origin and Host headers 
+in the Teleport configuration file for each application.
+
+To fix CSRF or CORS issues:
+
+1. Open the `/etc/teleport.yaml` file in a text editor.
```
Comment (zmb3) [inline docs/pages/application-access/troubleshooting-apps.mdx:42]: We should be explicit that this is the teleport.yaml file for the app service(s) that are proxying the app in question.

**Review by zmb3 — APPROVED** [review]
One comment, but otherwise looks great.

We might also want to mention that the certificate presented by the proxy must be trusted, that's caused lots of confusion in the past when using self-signed certificates.

File: docs/pages/application-access/troubleshooting-apps.mdx [hunk 2]
```diff
@@ -0,0 +1,59 @@
+---
+title: Troubleshooting Application Access
+description: Describes common issues and solutions for access to applications.
```
Comment (ptgott) [inline docs/pages/application-access/troubleshooting-apps.mdx:3]:
```suggestion
description: Describes common issues and solutions for access to applications proxied by Teleport.
```
Or something else that makes it clear that these are applications protected by Teleport.

File: docs/pages/application-access/troubleshooting-apps.mdx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+In most cases, you can fix these types of issues by adding explicit `rewrite` settings for the Origin and Host headers 
+in the Teleport configuration file for each application.
+
+To fix CSRF or CORS issues:
+
+1. Open the `/etc/teleport.yaml` file in a text editor.
+
+1. Add a `rewrite.headers` section similar to the following `grafana` example:
+
+   ```code
```
Comment (ptgott) [inline docs/pages/application-access/troubleshooting-apps.mdx:46]:
```suggestion
   ```yaml
```

**Review by ptgott — APPROVED** [review]

**Review by webvictim — APPROVED** [review]

File: docs/pages/application-access/troubleshooting-apps.mdx (hunk 1 above)
Comment (lsgunn-teleport) [reply docs/pages/application-access/troubleshooting-apps.mdx:42]: @webvictim, so would the fix be the same but just in a different file? For example (remove [ ] and add the rewrite.headers section):
47   apps:
       - name: grafana
         uri: http://localhost:3000/
         public_addr: [grafana.teleport.example.com](http://grafana.teleport.example.com/)
         rewrite:
             headers:
                Origin: "https://grafana.teleport.example.com/" # should be the Teleport application subdomain prepended with "https://"
                Host: "[grafana.teleport.example.com](http://grafana.teleport.example.com/)" # should be the Teleport application subdomain itself

**Comment by lsgunn-teleport** [comment]
@zmb3 What error do people see when using a self-signed or untrusted certificate?

**Pull request merged**